external/builder/babel-plugin-pdfjs-preprocessor.mjs

```diff
@@ -0,0 +1,241 @@
+import { types as t, transformSync } from "@babel/core";
+import fs from "fs";
+import { join as joinPaths } from "path";
+import vm from "vm";
+
+const PDFJS_PREPROCESSOR_NAME = "PDFJSDev";
+const ROOT_PREFIX = "$ROOT/";
+
+function isPDFJSPreprocessor(obj) {
+  return obj.type === "Identifier" && obj.name === PDFJS_PREPROCESSOR_NAME;
+}
+
+function evalWithDefines(code, defines) {
+  if (!code || !code.trim()) {
+    throw new Error("No JavaScript expression given");
+  }
+  return vm.runInNewContext(code, defines, { displayErrors: false });
+}
+
+function handlePreprocessorAction(ctx, actionName, args, path) {
+  try {
+    const arg = args[0];
+    switch (actionName) {
+      case "test":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("No code for testing is given");
+        }
+        return !!evalWithDefines(arg.value, ctx.defines);
+      case "eval":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("No code for eval is given");
+        }
+        const result = evalWithDefines(arg.value, ctx.defines);
+        if (
+          typeof result === "boolean" ||
+          typeof result === "string" ||
+          typeof result === "number" ||
+          typeof result === "object"
+        ) {
+          return result;
+        }
+        break;
+      case "json":
+        if (!t.isStringLiteral(arg)) {
+          throw new Error("Path to JSON is not provided");
+        }
+        let jsonPath = arg.value;
+        if (jsonPath.indexOf(ROOT_PREFIX) === 0) {
```

Comment:
This is obviously a pre-existing issue, but could we change this as follows?
```suggestion
        if (jsonPath.startsWith(ROOT_PREFIX)) {
```

(We have ESLint rules that *should* catch these cases, but given that ESLint is a static code analyser I'm guessing that it cannot safely tell that `jsonPath` is a string.)